From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Eduardo Lima (Etrunko)" <etrunko@redhat.com>
Date: Thu, 3 Aug 2017 16:53:09 -0300
Subject: [PATCH] Add missing #include in govirt.h

Signed-off-by: Eduardo Lima (Etrunko) <etrunko@redhat.com>
---
 govirt/govirt.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/govirt/govirt.h b/govirt/govirt.h
index fb7756f..9c92318 100644
--- a/govirt/govirt.h
+++ b/govirt/govirt.h
@@ -25,8 +25,11 @@
 #include <govirt/ovirt-enum-types.h>
 #include <govirt/ovirt-api.h>
 #include <govirt/ovirt-cdrom.h>
+#include <govirt/ovirt-cluster.h>
 #include <govirt/ovirt-collection.h>
+#include <govirt/ovirt-data-center.h>
 #include <govirt/ovirt-error.h>
+#include <govirt/ovirt-host.h>
 #include <govirt/ovirt-options.h>
 #include <govirt/ovirt-proxy.h>
 #include <govirt/ovirt-resource.h>
